test/functional/apps/visualize/index.js

```diff
@@ -7,14 +15,24 @@ import { bdd, remote, common, defaultTimeout, scenarioManager } from '../../../s
     bdd.before(function () {
       var self = this;
       remote.setWindowSize(1200,800);
-      // load a set of makelogs data
-      common.debug('loadIfEmpty logstashFunctional ' + self.timeout);
-      return scenarioManager.loadIfEmpty('logstashFunctional');
-    });
-
 
-    bdd.after(function unloadMakelogs() {
-      return scenarioManager.unload('logstashFunctional');
+      common.debug('Starting visualize before method');
+      var logstash = scenarioManager.loadIfEmpty('logstashFunctional');
+      return esClient.delete('.kibana')
+      .then(function () {
+        return common.try(function () {
+          return esClient.updateConfigDoc({'dateFormat:tz':'UTC', 'defaultIndex':'logstash-*'});
+        });
+      })
+      .then(function loadkibanaIndexPattern() {
+        common.debug('load kibana index with default index pattern');
+        return elasticDump.elasticLoad('visualize','.kibana');
+      })
+      // wait for the logstash data load to finish if it hasn't already
+      .then(function () {
+        return logstash.then();
```

Comment:
it should be sufficient to `return logstash` here and in _dashboard.js, these will both return a promise